File: hrcinfra/public/js/project_custom.js
// frappe.ui.form.on('Project', {
//     refresh: function(frm) {
//         console.log('Project form refreshed');
//     },
    
//     before_save: function(frm) {
//         console.log('Before saving Project');
        
//         if (frm.is_new()) {
//             console.log('This is a new Project, will create Warehouse');
            
//             frappe.call({
//                 method: 'frappe.client.insert',
//                 args: {
//                     doc: {
//                         doctype: 'Warehouse',
//                         warehouse_name: frm.doc.project_name
//                         // Add any other required fields for Warehouse
//                     }
//                 },
//                 callback: function(r) {
//                     console.log('Response from warehouse creation:', r);
//                     if (r.message) {
//                         frappe.show_alert({
//                             message: __('Warehouse created successfully'),
//                             indicator: 'green'
//                         });
//                     } else {
//                         console.error('Failed to create Warehouse');
//                         frappe.show_alert({
//                             message: __('Failed to create Warehouse'),
//                             indicator: 'red'
//                         });
//                     }
//                 }
//             });
//         } else {
//             console.log('This is an existing Project, not creating Warehouse');
//         }
//     }
// });

File: hrcinfra/public/py/project_custom.py
import frappe


def after_insert(doc, method):

# get_doc is used for two purposed, to fetch existing documents or create a new document
# When you pass a dictionary of values with the 'doctype' key, 
# it creates a new document instance but does not save it until you call .insert() or .save()
    parent_warehouse_doc= frappe.get_doc({
        'doctype':'Warehouse',
        'warehouse_name':doc.project_name,
        'custom_project':doc.name, 
        'is_group': 1,
    })

    parent_warehouse_doc.insert()

    
    store_warehouse_doc= frappe.get_doc({
        'doctype':'Warehouse',
        'warehouse_name':f"{doc.project_name}-Store",
        'custom_project':doc.name, 
        'parent_warehouse': parent_warehouse_doc.name
    })

    store_warehouse_doc.insert()



    frappe.msgprint("Warehouse Successfully Created")